src/crimson/os/seastore/seastore.h

```diff
@@ -200,12 +200,46 @@ class SeaStore final : public FuturizedStore {
   }
 
 
+  template <typename Ret, typename F>
+  auto repeat_with_onode(
+    CollectionRef ch,
+    const ghobject_t &oid,
+    F &&f) const {
+    return seastar::do_with(
+      oid,
+      Ret{},
+      TransactionRef(),
+      OnodeRef(),
+      std::forward<F>(f),
+      [=](auto &oid, auto &ret, auto &t, auto &onode, auto &f) {
+	return repeat_eagain([&, this] {
+	  t = transaction_manager->create_transaction();
+	  return onode_manager->get_onode(
+	    *t, oid
+	  ).safe_then([&, this](auto onode_ret) {
+	    onode = std::move(onode_ret);
+	    return f(*t, *onode);
+	  }).safe_then([&ret](auto _ret) {
+	    ret = _ret;
+	  });
+	}).safe_then([&ret] {
+	  return seastar::make_ready_future<Ret>(ret);
+	});
+      });
+  }
+
+
   friend class SeaStoreOmapIterator;
   omap_get_values_ret_t omap_list(
     CollectionRef ch,
     const ghobject_t &oid,
     const std::optional<string> &_start,
     OMapManager::omap_list_config_t config);
+  OMapManager::omap_list_ret __omap_list(
```

Comment:
please avoid starting identifiers with two underscores, these names are reserved for C/C++ compiler and standard library.

Reply:
done:-)

Reply:
should be fixed in #41222